types/models.d.ts

```diff
@@ -382,7 +382,7 @@ declare module 'mongoose' {
     validate(optional: any, pathsToValidate: PathsToValidate): Promise<void>;
 
     /** Watches the underlying collection for changes using [MongoDB change streams](https://www.mongodb.com/docs/manual/changeStreams/). */
-    watch<ResultType extends mongodb.Document = any>(pipeline?: Array<Record<string, unknown>>, options?: mongodb.ChangeStreamOptions & { hydrate?: boolean }): mongodb.ChangeStream<ResultType>;
+    watch<ResultType extends mongodb.Document = any, ChangeType extends mongodb.ChangeStreamDocument = any>(pipeline?: Array<Record<string, unknown>>, options?: mongodb.ChangeStreamOptions & { hydrate?: boolean }): mongodb.ChangeStream<ResultType, ChangeType>;
```

Comment:
Wouldn't it be better if we added a separate overload for `watch()` with `options: mongodb.ChangeStreamOptions & { fullDocument: 'updateLookup', hydrate?: boolean }` that added `fullDocument` to the result? It looks like this approach relies on the user to specify the correct type

Reply:
From my understanding of the [changelog](https://github.com/mongodb/node-mongodb-native/releases/tag/v4.6.0), the added feature is that the dev can now specify `ChangeType`, the type of the change document.
It is not related to the `options` parameter (I wrote an unclear example above, now edited).

Reply:
I took a closer look and @lpizzinidev is correct, the expected behavior here is that the user can overwrite the valid change types. `fullDocument: 'updateLookup'` doesn't always ensure that there will be a `fullDocument` property, for example deletes. Below is an example script I used to verify:

```ts
import mongoose from 'mongoose';

mongoose.set('debug', true);

run().catch(err => console.error(err));

async function run() {
  await mongoose.connect('mongodb://localhost:27017,localhost:27018,localhost:27019/mongoose_test');
  await mongoose.connection.dropDatabase();

  const TestModel = mongoose.model('Test', new mongoose.Schema({ name: String }));
  await TestModel.createCollection();

  await TestModel.
    watch<{name:string}>([], { fullDocument: 'updateLookup' }).
    on('change', change => console.log(change));
  await TestModel.create({ name: 'foo' });
  await TestModel.updateOne({ name: 'foo' }, { name: 'bar' });
  await TestModel.deleteOne({ name: 'bar' });
  console.log('Wrote document');
}
```